training/deepspeech_training/train.py

```diff
@@ -677,6 +680,303 @@ def __call__(self, progress, data, **kwargs):
         log_info('FINISHED optimization in {}'.format(datetime.utcnow() - train_start_time))
     log_debug('Session closed.')
 
+def train_with_horovod():
```

Comment:
I really don't think we should duplicate all this code, it's a recipe for problems.

Reply:
And honestly, it makes it very hard to review ...

Reply:
As mentioned in the PR description. I can merge it. It was just to keep it clear to your first review.

Reply:
I really think it'll be easier if you merge, sorry for the hassle.